mobile-app/app/screens/AppNavigator/screens/Dex/components/PoolPairCards/PoolPairCards.tsx

```diff
@@ -375,6 +185,245 @@ export function PoolPairCards ({
   )
 }
 
+interface PoolCardProps {
+  item: DexItem<WalletToken | PoolPairData>
+  expandedCardIds: string[]
+  setExpandedCardIds: (ids: string[]) => void
+  isFavouritePoolpair: (id: string) => boolean
+  setFavouritePoolpair: (id: string) => void
+  onAdd: (data: PoolPairData) => void
+  onRemove: (data: PoolPairData) => void
+  onSwap: (data: PoolPairData) => void
+  type: 'your' | 'available'
+  index: number
+}
+
+const PoolCard = ({
+  index,
+  item,
+  expandedCardIds,
+  isFavouritePoolpair,
+  setFavouritePoolpair,
+  type,
+  onAdd,
+  onRemove,
+  onSwap,
+  setExpandedCardIds
+}: PoolCardProps): JSX.Element => {
+  const { getTokenPrice } = useTokenPrice()
+  const { calculatePriceRates } = useTokenBestPath()
+  const { poolpairs: pairs } = useSelector((state: RootState) => state.wallet)
+  const blockCount = useSelector((state: RootState) => state.block.count)
+  const { data: yourPair } = item
+  const isFavouritePair = isFavouritePoolpair(yourPair.id)
+  const [priceRates, setPriceRates] = useState({
+    aToBPrice: new BigNumber(''),
+    bToAPrice: new BigNumber(''),
+    estimated: new BigNumber('')
+  })
+
+  const poolPairData = pairs.find(
+    (pr) => pr.data.symbol === (yourPair as AddressToken).symbol
+  )
+  const mappedPair = poolPairData?.data
+
+  useEffect(() => {
+    void getPriceRates()
+  }, [mappedPair, blockCount])
+
+  const getPriceRates = async (): Promise<void> => {
+    if (mappedPair !== undefined) {
+      const priceRates = await calculatePriceRates(
+        mappedPair?.tokenA.id,
```

Comment:
do we need the optional chaining `?.` here? 
```suggestion
        mappedPair.tokenA.id,
```